File: backend/src/test/java/com/skewwhiffy/auraltester/notation/model/scale/ScaleTypeTest.java
package com.skewwhiffy.auraltester.notation.model.scale;

import com.skewwhiffy.auraltester.notation.model.interval.DirectedInterval;
import com.skewwhiffy.auraltester.test.util.TestData;
import lombok.val;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.util.List;
import java.util.stream.IntStream;

import static org.assertj.core.api.Assertions.assertThat;

class ScaleTypeTest {
    private List<DirectedInterval> ascendingIntervals;
    private List<DirectedInterval> descendingIntervals;
    private ScaleType scaleType;

    @BeforeEach
    void setUp() {
        val displayName = TestData.random().string();
        val abc = TestData.random().string();
        ascendingIntervals = IntStream.range(0, 6).mapToObj(it -> TestData.random().directedInterval()).toList();
        descendingIntervals = IntStream.range(0, 8).mapToObj(it -> TestData.random().directedInterval()).toList();
        scaleType = new ScaleType(displayName, abc, ascendingIntervals, descendingIntervals);
    }

    @Test
    void when_scaleIsAscending_then_returnsAscendingIntervals() {
        val expected = ascendingIntervals;

        val actual = scaleType.getIntervals(ScaleDirection.ASCENDING);

        assertThat(actual).isEqualTo(expected);
    }

    @Test
    void when_scaleIsDescending_then_returnsDescendingIntervals() {
        val expected = descendingIntervals;

        val actual = scaleType.getIntervals(ScaleDirection.DESCENDING);

        assertThat(actual).isEqualTo(expected);
    }
}

File: backend/src/test/kotlin/com/skewwhiffy/auraltester/notation/model/scale/ScaleTypeTest.kt
package com.skewwhiffy.auraltester.notation.model.scale

import com.skewwhiffy.auraltester.notation.model.interval.DirectedInterval
import com.skewwhiffy.auraltester.test.util.TestData
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

internal class ScaleTypeTest {
    private lateinit var ascendingIntervals: List<DirectedInterval>
    private lateinit var descendingIntervals: List<DirectedInterval>
    private lateinit var scaleType: ScaleType

    @BeforeEach
    fun setUp() {
        val displayName = TestData.random.string
        val abc = TestData.random.string
        ascendingIntervals = (1..6).map { TestData.random.directedInterval }
        descendingIntervals = (1..8).map { TestData.random.directedInterval }
        scaleType = ScaleType(displayName, abc, ascendingIntervals, descendingIntervals)
    }

    @Test
    fun when_scaleIsAscending_then_returnsAscendingIntervals() {
        val actual = scaleType.getIntervals(ScaleDirection.ASCENDING)
        assertThat(actual).isEqualTo(ascendingIntervals)
    }

    @Test
    fun when_scaleIsDescending_then_returnsDescendingIntervals() {
        val actual = scaleType.getIntervals(ScaleDirection.DESCENDING)
        assertThat(actual).isEqualTo(descendingIntervals)
    }
}
